src/MassTransit.RabbitMqTransport/Topology/RabbitMqExchangeBindingExtensions.cs

```diff
@@ -63,9 +67,9 @@ public static ExchangeBindingSettings GetExchangeBinding(this SendSettings setti
 
         public static IEnumerable<ExchangeBindingSettings> GetExchangeBindings(this ReceiveSettings settings, string exchangeName)
         {
-            var exchange = new Exchange(exchangeName, settings.Durable, settings.AutoDelete, settings.ExchangeType);
-
-            var binding = new ExchangeBinding(exchange);
+            var exchange = new Exchange(exchangeName, settings.Durable, settings.AutoDelete, settings.ExchangeTypeDeterminer);
+            //TODO find a way for this to stay working while you want to implement routingkey structure

```

Comment:
Unclear what you need to on the next line. Do you want the `settings.RouteKeyFormatter.createRoutingKeyForType(type)` to be optional? Or do you want have a collection of formatters so the default one always works even when you override?